### PR TITLE
override Range#{tails, inits}

### PR DESCRIPTION
**Pull request by xuwei-k**

### CODE REVIEW

File: src/library/scala/collection/immutable/Range.scala [hunk 1]
```diff
@@ -354,6 +354,35 @@ sealed abstract class Range(
       else head
     } else super.max(ord)
 
+  override def tails: Iterator[Range] =
```
Comment (joshlemer) [inline src/library/scala/collection/immutable/Range.scala:357]: Smart!

**Review by joshlemer — APPROVED** [review]

File: src/library/scala/collection/immutable/Range.scala (hunk 1 above)
Comment (viktorklang) [reply src/library/scala/collection/immutable/Range.scala:357]: Shouldn't `tails` return the empty iterator for an empty range?

File: src/library/scala/collection/immutable/Range.scala [hunk 2]
```diff
@@ -354,6 +354,35 @@ sealed abstract class Range(
...
+          val res = Range.this.drop(i)
+          i += 1
+          res
+        } else {
+          Iterator.empty.next()
+        }
+      }
+    }
+
+  override def inits: Iterator[Range] =
```
Comment (viktorklang) [inline src/library/scala/collection/immutable/Range.scala:372]: Same here.

File: src/library/scala/collection/immutable/Range.scala (hunk 1 above)
Comment (xuwei-k) [reply src/library/scala/collection/immutable/Range.scala:357]: I don't change behavior.

```
Welcome to Scala 2.12.6 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_181).
Type in expressions for evaluation. Or try :help.

scala> Nil.tails.toList
res0: List[List[Nothing]] = List(List())

scala> (1 until 1).tails.toList
res1: List[scala.collection.immutable.IndexedSeq[Int]] = List(Vector())
```

```
Welcome to Scala 2.13.0-M5 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_181).
Type in expressions for evaluation. Or try :help.

scala> Nil.tails.toList
res0: List[List[Nothing]] = List(List())

scala> (1 until 1).tails.toList
res1: List[IndexedSeq[Int]] = List(Vector())
```

Comment (viktorklang) [reply src/library/scala/collection/immutable/Range.scala:357]: @xuwei-k I stand corrected, Yoshida-san! :)

**Review by szeiger — APPROVED** [review]

**Pull request merged**